specs/Booking_spec.rb

```diff
@@ -0,0 +1,306 @@
+
+      new_hotel.make_block(check_in,check_out,rooms_in_block,10)
+
+      new_hotel.all_blocks.count.must_equal 1
+
+      new_block = new_hotel.all_blocks[0]
+      block_id = new_block.id
+
+      new_hotel.make_block_reservation(block_id,room_id)
+
+      proc {new_hotel.make_block_reservation(block_id,room_id)}.must_raise ArgumentError
+
+      new_block_reservation = new_hotel.all_reservations[-1]
+      new_block_reservation.block_id.must_equal "B1"
+      new_block_reservation.type.must_equal :block
+    end
+
+  end
+
+  describe "make_block" do
+    it "creates a Block and reserves a room" do
+      #note, need to fix: it shouldn't reserve room officially, but should be "Blocked off" test for this when the functionality is made
+
+      new_hotel =  HotelBooking::Hotel.new
+
+      new_hotel.make_block("2018-03-14", "2018-03-16",[18,19,20], 2)
+
+      new_block = new_hotel.all_blocks[0]
+
+
+      new_block.must_be_instance_of HotelBooking::Block
+      new_block.id.must_equal "B1"
+
+      new_block.room_ids.each do |room_id|
+        room = new_hotel.find_room_by_id(room_id)
+        (Date.parse("2018-03-14")...Date.parse("2018-03-16")).each do |date|
+          room.all_dates.must_include date #fix this logic later
+        end
+      end
+    end
+
+    it "does not allow you to create a Block if the room is reserved for any of the dates within the range you've proposed" do
+      new_hotel =  HotelBooking::Hotel.new
+
+      check_in= "2018-04-03"
+      check_out = "2018-04-17"
+      room_id = 5
+      new_hotel.make_reservation(check_in,check_out,room_id)
+
+      room = new_hotel.find_room_by_id(room_id)
+
+      (Date.parse(check_in)...Date.parse(check_out)).to_a.each do |date|
+        room.all_dates.must_include date
+      end
+
+      block_start= "2018-04-01"
+
+      proc{new_hotel.make_block(block_start,check_in,[1,5],10)}.must_raise ArgumentError
+
+      block_start2 = "2018-04-05"
+      block_end = "2018-04-08"
+
+      proc{new_hotel.make_block(block_start2,block_end,[1,5],10)}.must_raise ArgumentError
+
+      block_end2 = "2018-05-01"
+      new_hotel.make_block(check_out,block_end2,[1,5],10)
+
+    end
+  end
+
+
+  describe "available_rooms" do
+    before do
+      @new_hotel = HotelBooking::Hotel.new
+      @check_in = Date.parse("June 1, 2018")
+      @check_out = Date.parse("June 30, 2018")
+      @june_rooms = @new_hotel.available_rooms(@check_in, @check_out)
+    end
+
+    it "returns an array of available rooms" do
+
+      @june_rooms.must_be_instance_of Array
+      @june_rooms.count.must_equal 20
+      @june_rooms.each do |room|
+        room.must_be_instance_of HotelBooking::Room
+      end
+    end
+
+    it "updates correctly when all rooms are reserved" do
+      @new_hotel.all_rooms.each do |room|
+        guest_idx = 1
+        res_idx = 10
+        room.reserve_room(@check_in, @check_out, res_idx, guest_idx)
+
+        guest_idx +=1
+        res_idx +=1
```

Comment:
at each iteration, because you're resetting the values of `guest_idx` and `res_idx` in lines 164 and 165, incrementing them here doesn't actually affect the code

Your tests pass because you aren't testing for this logic/asserting that the guest and res ids are set!